### PR TITLE
Support operations (e.g. EQUALS) on MINPVV

### PR DESCRIPTION
**Pull request by blattms**
With this change MINPVV behaves like a regular field property and e.g. setting and or changing it via EQUALS.

Before this change MINPVV needed to be defined directly using the MINPVV keyword and changes applied to it later where silently ignored and lost.

Removed cells look good in the eye-norm at least.

### CODE REVIEW

File: opm/input/eclipse/EclipseState/Grid/FieldProps.hpp [hunk 1]
```diff
@@ -131,6 +131,7 @@ namespace ALIAS {
 
 namespace GRID {
 static const std::unordered_map<std::string, keyword_info<double>> double_keywords = {{"DISPERC",keyword_info<double>{}.unit_string("Length")},
+                                                                                      {"MINPVV",  keyword_info<double>{}.unit_string("ReservoirVolume")},
```
Comment (blattms) [inline opm/input/eclipse/EclipseState/Grid/FieldProps.hpp:134]: I would hope that the unit_string is not needed and is implied by the json definition of MINPVV

**Comment by blattms** [comment]
Note that the MINPVV array will now be stored twice. Maybe we should try to save some memory later (also for the case where MINPV is used and we currently store a full vector, too).

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
Also note that I have not changed precedence of  MINPV over MINPORV, which has precedence over MINPVV.

**Comment by blattms** [comment]
I am in the process of adding a test. Something strange is happening there...

**Comment by blattms** [comment]
Added a unit test (quite some surprises there in this part of the code) and now we remove MINPVV from the field properties again once it is not needed anymore.

**Comment by blattms** [comment]
There a bit too many surprises here for my taste.

I wanted to move setting of the minpv vector in the EclipseGrid before 
```
this->m_inputGrid.resetACTNUM(this->field_props.actnum());
this->field_props.reset_actnum(this->getInputGrid().getACTNUM());
```
because then MINPVV would not need to be a global array anymore.

To my surprise this caused test failures for the test where MINPV is set only via
```
EQUALS
    MINPVV 100.0 1* 1* 2 2 1 1 /
```
because `FieldPropsManager::has_double` return `false` for MINPV there. Apparently the resetting of actnum does a lot of additional magic.

I prefer to leave this PR as  is and am undrafting it again...

**Comment by blattms** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
> Apparently the resetting of actnum does a lot of additional magic.

Anything that changes the set of active cells is painful!

**Comment by blattms** [comment]
Well the magic goes like this:
- Query PORV uncondtionally (and create the array if it is not there)
- Use it to set ACTUM to zero where PORV is 0.
- compress all field properties (no that MINPV is one it would be compressed, too).

Well, now we use uncompressed MINPV before resetActnum. Still, creating a PORV is a bit bothering.

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
rebased this one and will run jenkins again

**Comment by blattms** [comment]
jenkins build this please

File: opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp [hunk 2]
```diff
@@ -1936,6 +1933,13 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
         }
     }
 
+    void EclipseGrid::setMINPVV(const std::vector<double>& minpvv) {
+        if (m_minpvMode == MinpvMode::Inactive) {
+            assert(minpvv.size() == this->getCartesianSize());
```
Comment (akva2) [inline opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp:1938]: assert on user provided data.

File: opm/input/eclipse/EclipseState/Grid/FieldProps.hpp [hunk 3]
```diff
@@ -541,6 +541,7 @@ class FieldProps {
 
     std::vector<std::string> fip_regions() const;
 
+    void deleteMINPVV();
```
Comment (akva2) [inline opm/input/eclipse/EclipseState/Grid/FieldProps.hpp:544]: space before private

File: tests/parser/EclipseGridTests.cpp [hunk 4]
```diff
@@ -944,10 +944,9 @@ BOOST_AUTO_TEST_CASE(MinPVV) {
     Opm::EclipseState es( deck);
     const auto& fp = es.fieldProps();
     const auto& grid = es.getInputGrid();
-    const auto& fp_minpvv = fp.get_global_double("MINPVV");
+    std::vector<double> fp_minpvv = {1000, 1000, 1000, 1010, 1010, 1010};
```
Comment (akva2) [inline tests/parser/EclipseGridTests.cpp:946]: please use doubles and not ints.

File: opm/input/eclipse/EclipseState/Grid/FieldProps.hpp (hunk 3 above)
Comment (blattms) [reply opm/input/eclipse/EclipseState/Grid/FieldProps.hpp:544]: I don't know what you mean by this

File: opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp (hunk 2 above)
Comment (blattms) [reply opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp:1938]: Sorry I am too dense. What am I supposed to do here?

Comment (blattms) [reply opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp:1938]: Maybe it was too early for me. Got it.

**Comment by blattms** [comment]
Thanks a lot for the review (I am awake now :sweat_smile:). Comments should be addressed now. This is also tested by Client now.

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
jenkins build this please

**Review by akva2 — APPROVED** [review]
Looks good to me, has tests, is battle tested. Go for it.

**Pull request merged**